### DISCUSSION TITLE
Welcome to client-zip Discussions!

### DISCUSSION THREAD

**Discussion by Touffy** [Touffy/client-zip#63] [Announcements]
## 👋 Welcome!

Discussions are here for users of client-zip to ask about client-zip usage, share interesting use-cases, and other things that are not really "issues". Please keep opening issues for bug reports and feature requests.

I might also use Discussions for announcements, examples, polls…

So, that's it. Let's see what happens.

**Comment by Andinh123** [comment]
Hey! The client-zip works really well. But how can I include a file which uploaded by the user themselve into the zip folder? I'm trying to make a tools that the user uploads a file, then JS will fetch certain files based on the uploaded file from my server, finally it will zip them all up. But I can't find any doc about adding uploaded file.

**Reply by Touffy** [reply]
The documentation does say that you can use `File` objects as input.

Your user sends the file using an `<input type="file">` in a form, right ? just access the `files` property on that input. It's an array of `File` objects (in your case, it will only have one item). And of course you can mix Files and Responses in `downloadZip`. The File can be reused as many times as you like — both for sending it to your back-end for processing, and copying it in `downloadZip`.

**Reply by Touffy** [reply]
The MDN page for `<input type="file">` has [a thorough example](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#examples) on how to get Files out of it.

I guess, in your use case, the trickiest part is not getting the File, but how you handle getting more files to go with the user's file (where is the logic ? can some of it be moved client-side ?). I cannot easily write relevant code for that because I don't know what you actually need, or your constraints, or the performance expectations. If it's meant to be an open-source tool, could you publish the source ?
